File: pdu/python/hako_msgs/pdu_cdr_conv_DroneStatus.py
from .pdu_pytype_DroneStatus import DroneStatus
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_DroneStatus(writer: CdrWriter, src: DroneStatus):
    writer.write_int32(src.flight_mode)
    writer.write_int32(src.internal_state)
    py_to_cdr_body_Vector3(writer, src.propeller_wind)
    writer.write_int32(src.collided_counts)


def cdr_body_to_py_DroneStatus(reader: CdrReader, dst: DroneStatus):
    dst.flight_mode = reader.read_int32()
    dst.internal_state = reader.read_int32()
    cdr_body_to_py_Vector3(reader, dst.propeller_wind)
    dst.collided_counts = reader.read_int32()
    return dst


def py_to_cdr_DroneStatus(src: DroneStatus) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_DroneStatus(writer, src)
    return writer.bytes()


def cdr_to_py_DroneStatus(cdr_payload) -> DroneStatus:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = DroneStatus()
    return cdr_body_to_py_DroneStatus(reader, dst)

File: pdu/javascript/hako_msgs/pdu_cdr_conv_DroneStatus.js
import { DroneStatus } from './pdu_jstype_DroneStatus.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduDroneStatusConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { DroneStatus } src
     */
    static to_cdr_body(writer, src) {
        writer.write_int32(src.flight_mode);
        writer.write_int32(src.internal_state);
        PduVector3Converter.to_cdr_body(writer, src.propeller_wind);
        writer.write_int32(src.collided_counts);
    }

    /**
     * @param {PduCdrReader} reader
     * @param { DroneStatus } dst
     * @returns { DroneStatus }
     */
    static cdr_body_to_js(reader, dst) {
        dst.flight_mode = reader.read_int32();
        dst.internal_state = reader.read_int32();
        PduVector3Converter.cdr_body_to_js(reader, dst.propeller_wind);
        dst.collided_counts = reader.read_int32();
        return dst;
    }

    /**
     * @param { DroneStatus } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { DroneStatus }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new DroneStatus());
    }
}
